### ISSUE TITLE
Content Security Policy: eval blocked

### ISSUE THREAD

**Issue by scola84** [eligrey/FileSaver.js#491]
In Firefox 63+ (Ubuntu and Windows) and Edge I get errors like

```
Content Security Policy: The page’s settings blocked the loading of a resource at eval (“default-src”).
``` 
Is it possible to solve this in v2? Could you put the saveAs function in a seperate file in src? Then I can bypass the attachment of the function to global.

**Comment by jimmywarting** [comment]
I was hoping to avoid this with a try/catch

The affected part is this:
https://github.com/eligrey/FileSaver.js/blob/00183093bf92592fe231e56b0e681fe150123ccb/dist/FileSaver.js#L27-L36

**Comment by scola84** [comment]
Yeah, I understood from the comment. I don't know how old that part of the answer from Stackoverflow is, but apparently some browser have become even stricter. I'm currently trying some magic with Rollup to get rid of the definition of the global.

**Comment by WuglyakBolgoink** [comment]
same hier! Can reproduce in Firefox@windows, on MacOS Firefox show Error
```
Content Security Policy: Die Einstellungen der Seite haben das Laden einer Ressource auf eval blockiert ("script-src").
```
but content will be loaded correct.
Chrome@macos@latest - ok
yandexBrowser@macOS@latest - ok
Safari@macos@latest - ok
why this happen always at Friday?)

**Comment by WuglyakBolgoink** [comment]
up.

**Comment by scola84** [comment]
I solved it by replacing `Function('return this')() || (42, eval)('this');` with `window` during the `rollup` building process.

Is this package ever used in any other context than the browser?

Anyway, I think you should export your core function in a separate file and attach it to any other context in specific files for those context. For me, this issue can be closed.

**Comment by jimmywarting** [comment]
FileSaver don't work in any other context but the window. But some includes it in web workers and nodejs for some reason

The solution would be to just remove the try and only use the cache method
```js
 var _global = typeof window === 'object' && window.window === window 
  ? window : typeof self === 'object' && self.self === self 
  ? self : typeof global === 'object' && global.global === global 
  ? global 
  : this;
```

**Comment by jimmywarting** [comment]
released as v2.0.0
